parkbenchcommon/broadcastconsumer.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2017-2018 Joel Allen Luellwitz and Andrew Klapp
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Provides the consuming component of a filesystem-based IPC mechanism."""
+
+import os
+import logging
+import time
+import tmpfs
+
+__all__ = ['BroadcastCheckError', 'BroadcastConsumer']
+
+BROADCAST_PATH = '/var/spool/'
+TMPFS_SIZE = '1M'
+
+class BroadcastCheckError(Exception):
+    """ This exception is raised when an error is encountered while attempting to check a
+    broadcast.
+    """
+
+class BroadcastConsumer(object):
+    """ Provides the consuming component of a filesystem-based IPC mechanism.
+
+    program_name: The name of the program that broadcasts this broadcast
+    broadcast_name: The name of this broadcast
+    check_intveral: The number of seconds to wait in between re-reading the filesystem
```

Comment:
'check_interval' is misspelled. minimum_delay might be more appropriate.

Suggested text for this parameter is "The minimum delay in seconds between broadcasts. If a second broadcast occurs before minimum_delay has passed, the second broadcast is ignored.".